packages/fetchai/skills/ml_train/handlers.py

```diff
@@ -129,36 +129,25 @@ def _handle_terms(
         :param ml_trade_dialogue: the dialogue object
         :return: None
         """
-        terms = ml_trade_msg.terms
+        proposal_terms = ml_trade_msg.terms
         self.context.logger.info(
             "received terms message from {}: terms={}".format(
-                ml_trade_msg.sender[-5:], terms.values
+                ml_trade_msg.sender[-5:], proposal_terms.values
             )
         )
 
         strategy = cast(Strategy, self.context.strategy)
-        acceptable = strategy.is_acceptable_terms(terms)
-        affordable = strategy.is_affordable_terms(terms)
+        acceptable = strategy.is_acceptable_terms(proposal_terms)
+        affordable = strategy.is_affordable_terms(proposal_terms)
         if not (acceptable and affordable):
             self.context.logger.info(
                 "rejecting, terms are not acceptable and/or affordable"
             )
             return
 
         if strategy.is_ledger_tx:
-            # construct a tx for settlement on the ledger
-            ledger_api_dialogues = cast(
-                LedgerApiDialogues, self.context.ledger_api_dialogues
-            )
-            terms_ = strategy.terms_from_proposal(ml_trade_msg.terms)
-            ml_trade_dialogue.terms = terms_
-            _, ledger_api_dialogue = ledger_api_dialogues.create(
-                counterparty=LEDGER_API_ADDRESS,
-                performative=LedgerApiMessage.Performative.GET_RAW_TRANSACTION,
-                terms=terms_,
-            )
-            ledger_api_dialogue = cast(LedgerApiDialogue, ledger_api_dialogue)
-            ledger_api_dialogue.associated_ml_trade_dialogue = ml_trade_dialogue
```

Comment:
What happened to that? Why can it be removed?

Reply:
Not sure why this block was there. 

This logic (sending GET_RAW_TRANSACTION) is what happens when we add the ml_dialogue to the tx_behaviour for processing (two lines below).

Reply:
Well spotted!